client/src/util/dataframe/types.ts

```diff
@@ -66,6 +70,11 @@ export interface DataframeColumn extends DataframeColumnGetter {
    */
   isContinuous: boolean;
 
+  /**
+   * Boolean indicating if the underlying data is dictionary-encoded.
+   */
+  isCategorical: boolean;
```

Comment:
this is very, very confusing as it is a different use of the word `categorical` than the other interfaces.  Eg, `histogramCategorical` does not require a dictionary-encoded column (it works on any array/column type).

I would come up with a new term, and not overload the concept - which will lead to much confusion.

(ie, clean abstractions!)

Reply:
Another thought - I think you may want to create multiple specialized sub-interfaces for DataframeColumn, so it can be used as a "type", eg,
* CategoricalDataframeColumn
* NumericDataframeColumn
* etc...
thoughts?

Reply:
I think that's a great idea. It'd let us explicitly define attributes specific to categorical vs numerical columns.

Reply:
I think I need a primer on how types work to resolve this properly...